discovery_engine_core/tooling/src/bin/backend/routes.rs

```diff
@@ -0,0 +1,249 @@
+// Copyright 2022 Xayn AG
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, version 3.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+use actix_web::{
+    get,
+    post,
+    web::{Data, Json, Query},
+    Responder,
+};
+use log::debug;
+use reqwest::Client;
+use serde_json::{json, Value};
+
+use crate::{
+    elastic::{Hit, MindArticle, Response},
+    errors::BackendError,
+    newscatcher::{Article as NewscatcherArticle, Response as NewscatcherResponse},
+    AppState,
+    Config,
+    SearchParams,
+};
+
+#[get("/search")]
+pub(crate) async fn search_get(
+    config: Data<Config>,
+    app_state: Data<AppState>,
+    client: Data<Client>,
+    search_params: Query<SearchParams>,
+) -> Result<impl Responder, BackendError> {
+    handle_search(&config, &app_state, &client, &search_params).await
```

Comment:
❓ can we also pass the args by reference? not sure how actix handles this internally, but since we only need them by ref, we can probably save some cloning. similarly for the other routes.
```suggestion
#[get("/search")]
pub(crate) async fn search_get(
    config: &Data<Config>,
    app_state: &Data<AppState>,
    client: &Data<Client>,
    search_params: &Query<SearchParams>,
) -> Result<impl Responder, BackendError> {
    handle_search(config, app_state, client, search_params).await
```

Reply:
tried your suggestion but unfortunately it appears no, e.g. changing to `search_params: &Query<SearchParams>`:
```
error[E0277]: the trait bound `&actix_web::web::Query<SearchParams>: actix_web::FromRequest` is not satisfied
   --> tooling/src/bin/backend/routes.rs:34:1
    |
34  | #[get("/search")]
    | ^^^^^^^^^^^^^^^^^ the trait `actix_web::FromRequest` is not implemented for `&actix_web::web::Query<SearchParams>`
    |
    = help: the trait `actix_web::FromRequest` is implemented for `actix_web::web::Query<T>`
    = note: required because of the requirements on the impl of `actix_web::FromRequest` for `(actix_web::web::Data<Config>, actix_web::web::Data<AppState>, actix_web::web::Data<reqwest::Client>, &actix_web::web::Query<SearchParams>)`
```

Reply:
ok, i see. seems to be caused by the move semantics which the `FromRequest` trait imposes.